### ISSUE TITLE
hash css classname without webpack setup

### ISSUE THREAD

**Issue by indraraj26** [angular/angular-cli#20621]
# 🚀 Feature request

### Command (mark with an `x`)

- [ ] new
- [x] build
- [x] serve
- [ ] test
- [ ] e2e
- [ ] generate
- [ ] add
- [ ] update
- [ ] lint
- [ ] extract-i18n
- [ ] run
- [ ] config
- [ ] help
- [ ] version
- [ ] doc

### Description
ng serve --hashCssClass
ng build --hashCssClass

### Describe the solution you'd like
 If you have a solution in mind, please describe it.
We will have configuration in angular.json there we can choose what we want. Modules: post css loader

### Describe alternatives you've considered
 Have you considered any alternative solutions or workarounds?
i see for react that they use eject and reconfigured the postcsss loader

**Comment by clydin** [comment]
Can you please provide additional background and use cases for the feature request?
Please also note that Angular components automatically support view encapsulation for styles.  For additional information regarding view encapsulation, please see https://angular.io/guide/view-encapsulation.

**Comment by indraraj26** [comment]
with that bundle size can be reduce lets i have class name

`.dashobard-container`

`.g-d-c`

example : https://www.google.com/amp/s/www.freecodecamp.org/news/reducing-css-bundle-size-70-by-cutting-the-class-names-and-using-scope-isolation-625440de600b/amp/


inspiration from all google site you cant say anything by reading the class name.

**Comment by alan-agius4** [comment]
Thanks for this request @indraraj26, 

The CSS class hashing is typically used to tackle CSS encapsulation, Angular already provides an out-of-box solution for this which doesn't require any additional tooling for it to work.

While in some cases, it will reduce the size of the CSS bundle, it's also important to mention that in many cases it will result in actually increasing the size of the CSS because something like;

```css
:local(.className) {
  background: red;
  color: yellow;
}

:local(.subClass) {
  composes: className;
  background: blue;
}
```

Will be converted to 
```css
._23_aKvs-b8bW2Vg3fwHozO {
  background: red;
  color: yellow;
}

._13LGdX8RMStbBE9w-t0gZ1 {
  background: blue;
}
```

This approach is also typically used with TSX/JS where you statically reference the CSS class into your JavaScript example;

file.css
```css
.button {
  font-size: 16px;
}
```

component.js
```jsx
import style from './file.css';

document.body.innerHTML =  `<div class=${style.app}></div>`;
```

In the above post you mentioned Google applications. The goal here is indeed to reduce the size and this is possible because some applications in Google use [Closure Stylesheets](https://github.com/google/closure-stylesheets) which is extension to CSS which makes CSS class renaming possible when used with [Closure Compiler](https://developers.google.com/closure/compiler/), that being said, AFAIK, Closure Stylesheets is being replaced with standard SASS inside Google.

To sum it up, indeed there are cases why renaming CSS classes can reduce the file size, but for this to work properly you need to write your code in a certain way and use specific tooling. I don't see this being part of Angular CLI roadmap anytime soon.
